### ISSUE TITLE
Jackett ignores -j (ProxyConnection) CLI flag

### ISSUE THREAD

**Issue by verstoff** [Jackett/Jackett#9467]
**OS**: docker container (https://hub.docker.com/r/linuxserver/jackett)

**Jackett Version**: CoreLinuxAmdx64 v0.16.1082.0

**Are you using a proxy or VPN?** yes

### Description

- Run Jackett from CLI with argument `-j, --ProxyConnection` specifying a valid HTTP proxy. Tested `http://proxyhost:3128`, `proxyhost:3128`, etc
- See in console:
```
09-07 20:50:12 Info Proxy enabled. http://proxyhost:3128 (cli arg successfully passed)
...
09-07 20:50:13 Info Using Proxy: No (but proxy setting not applied)
```
- Try to make some request, e.g. test indexer
- See error:
` An error occurred while testing this indexer
Exception (rutracker): Authentication failed because the remote party has closed the transport stream.: The SSL connection could not be established, see inner exception.`
- Setup same proxy but with Web UI
- Check that same test works

Not familiar with C# but look like there is no code in project that would use this variable:
https://github.com/Jackett/Jackett/search?q=ProxyConnection&unscoped_q=ProxyConnection

**Comment by ngosang** [comment]
You are right, this is not implemented.

**Comment by hoborm** [comment]
I've done some investigation around the CLI parameters and it seems that the configuration logic is a bit complicated and the CLI-side needs some extension for it to be able to properly configure proxy setup. eg. username, password. (Which is already present on the web ui) I was going down the rabbit hole to debug it and I think I narrowed down what the issue is. I can volunteer to fix this issue if it's still available. In the meanwhile you can still setup the proxy from the web @verstoff, if that's available for you.

**Comment by ngosang** [comment]
> I can volunteer to fix this issue if it's still available.

Sure it's. Any help is welcome.

**Comment by ngosang** [comment]
@garfield69 since this is not implemented and I don't want to do it anytime soon.. could we remove this flag from documentation and keep this issue as enhancement?

**Comment by garfield69** [comment]
```
C:\ProgramData\Jackett>jackettconsole --help
Jackett v0.17.145
  -i, --Install            Install Jackett windows service (Must be admin)
  -r, --ReserveUrls        (Re)Register windows port reservations (Required for listening on all interfaces).
  -u, --Uninstall          Uninstall Jackett windows service (Must be admin).
  -l, --Logging            Log all requests/responses to Jackett
  -t, --Tracing            Enable tracing
  -c, --UseClient          Override web client selection. [automatic(Default)/httpclient/httpclient2]
  -j, --ProxyConnection    use proxy - e.g. 127.0.0.1:8888
  -s, --Start              Start the Jacket Windows service (Must be admin)
  -k, --Stop               Stop the Jacket Windows service (Must be admin)
  -x, --ListenPublic       Listen publicly
  -z, --ListenPrivate      Only allow local access
  -p, --Port               Web server port
  -m, --MigrateSettings    Migrate settings manually (Must be admin on Windows)
  -n, --IgnoreSslErrors    [true/false] Ignores invalid SSL certificates
  -d, --DataFolder         Specify the location of the data folder (Must be admin on Windows) eg. --DataFolder="D:\Your
                           Data\Jackett\". Don't use this on Unix (mono) systems. On Unix just adjust the HOME directory of
                           the user to the datadir or set the XDG_CONFIG_HOME environment variable.
  --NoRestart              Don't restart after update
  --PIDFile                Specify the location of PID file
  --NoUpdates              Disable automatic updates
  --help                   Display this help screen.
  --version                Display version information.
```
not sure I understand.  you want to remove -j from the readme but keep the -j in the jackettconsole --help?

**Comment by ngosang** [comment]
I didn't remember the help code. I will remove both.

**Comment by francis-io** [comment]
I'm not sure that this was not working. For me, using a proxy connection did work. Please see my below, cut down, docker-compose example using `linuxserver/jackett`, but it should demonstrate what I'm seeing. This used to let me use the proxy provided by `haugene/transmission-openvpn` but now throws an error, since the `ProxyConnection` option was removed. Without this option set, almost half the remote apis return an error when tested. Up until recently, setting the ProxyConnection option made them all work and return correctly. 

```
services:
transmission-openvpn:
    image: haugene/transmission-openvpn
    container_name: transmission-openvpn
    environment:
      - WEBPROXY_ENABLED=true #allow other containers to use openvpn

  jackett:
    container_name: jackett
    image: "linuxserver/jackett:version-v0.17.153"
    environment:
      - RUN_OPTS=--ProxyConnection=transmission-openvpn:8888
```

**Comment by ngosang** [comment]
I'm sure that CLI option was not implemented.
Jackett still supports proxy, you have to remove the `ProxyConnection` for the docker-compose file and configure the proxy in the Jackett UI.

**Comment by francis-io** [comment]
Thank you. 

In my testing a while ago I must have had both options, the CLI and GUI set, but forgotten about it.

**Comment by hoborm** [comment]
I was able to create a POC with a hard-wired proxy connection type with a proxy where no authentication required. I have found that Jackett has two types of configuration structures one is the "regular" and the other is "runtime" configuration which supposedly should not save to the config file (Some of them still do as far as I remember). 

I can create an implementation where the proxy type could be passed-in with parameters however I could not find a good way to pass in the credentials needed for the proxy authentication. (_I'm open for suggestions_) On top of that the whole configuration (CLI, GUI) are a little bit hard to follow which raised a question whether we still need the "runtime" configuration. If we still need the runtime configuration I'd suggest to separate them completely because the startup and configuration introduced some unnecessary logic because of them while still representing them as a nested structure from the outside. 

I'm open for discussion on either solutions and happy to implement any them as I still would like to contribute I just wasn't sure on the way I should take.

**Comment by ngosang** [comment]
I removed this argument because it was not working. I closed this issue because I don't this it's useful to have it.

It's easy to write the remaining code and add more arguments for proxy user/pass. The main issue is the user/password will be in plain text. You will be able to see the pass in docker-compose.yml, docker ps, ps aux ....
When you configure the pass in the UI, it's saved encrypted, the same as the indexer credentials.

**Comment by mkrupczak3** [comment]
My server log brought me here. What did you monkeys break?

**Comment by mkrupczak3** [comment]
#10711 

Hm. I don't seem to remember setting my server to update with 'nightly' builds. 

...

Oh no, what's that?


You broke my Docker config on a stable branch? (ಠ_ಠ)

**Comment by mkrupczak3** [comment]
#9467 and #10711 Should both be re-opened, each are unique and perfectly valid. 

There's no reason for inconsequential, breaking changes like this to be made on a stable branch.

**Comment by ilike2burnthing** [comment]
The answer to your issue was already given:
> Jackett still supports proxy, you have to remove the `ProxyConnection` for the docker-compose file and configure the proxy in the Jackett UI.

Is there a reason this isn't suitable, beyond the 2 minutes of inconvenience of having to do so?

Please show a bit of common courtesy and respect towards those volunteering to maintain the free software you use. If you have an issue or suggestion, we're happy to listen and help.

**Comment by mkrupczak3** [comment]
Hi @ilike2burnthing,

Thank you for your patience, courtesy, and renewed interest in this issue. 

While it may seem like this is a quick fix for most administrators, this relies on the assumption that users of Jackett with Docker know where to look. 

This may be a flawed assumption, and may be more difficult for administrators to correct than it might seem.

I reckon just about anyone who has a Plex server setup with Docker uses Jackett. In addition, any container stack implementing Jackett ultimately depends upon its proper operation for privacy, security, and functionality.

Not only does this change break any likely Docker configuration in a way that is difficult to diagnose. The administrator must look through logs of all containers, and finally read Jackett's log throwing an error, interpret the nature of this error, and make a correction to their configuration. 

It is likely that many administrators with such a Docker Plex setup are novice administrators, using a config written by someone else, and/or otherwise incapable of understanding that a change made in a stable release of Jackett broke their config. 

It may as such be difficult for such novice users to correct. 


Also, this change breaks Jackett's ability to use a proxy when instantiated with Docker. In a Docker container stack, it is likely that Jackett is often used in association with a proxy-providing container. This removal of functionality, in a stable release, is a severe regression.

It may expose administrators to previously un-calculated privacy and security risks.

**Comment by mkrupczak3** [comment]
At the very least, this issue should remain open for Administrators' reference, and so that an acceptable resolution can be reached.

**Comment by ilike2burnthing** [comment]
Given that it didn't work before, if anything this is a massive improvement in terms of privacy and security, as at least now the user will get an error that the flag for their proxy is unknown, and they can look into the issue further. Whereas before, they thought they were setting up a proxy, actually weren't, but had no real way of knowing as everything still worked.

This issue being left open isn't really going to help anyone find it any easier, as it will quickly be buried under newer ones (and the one over on the ls.io repo won't be hard to find for a while - https://github.com/linuxserver/docker-jackett/issues/111).

Reference to the flag has been removed from documentation, this issue is easily found when searching 'proxyconnection', there is a error in the logs, proxy use is still available in the UI (with authentication as well); I'm not sure there's anything more to be done. If someone wants to submit a PR which properly implements it, then by all means, but we're of limited resources.

**Comment by flatline-studios** [comment]
Yup. Wasted an hour of my existence tracking down this issue.

Why make this error out the command, instead of just generating a warning if the option doesn't work?

Breaking a stable branch is a big no-no where I come from.
